packages/services/storage/src/index.ts

```diff
@@ -3574,6 +3576,25 @@ export async function createStorage(connection: string, maximumPoolSize: number)
               `
               : sql``
           }
+          ${
+            failed
+              ? sql`
+                AND (
+                  "is_success" = false
+                )
+              `
+              : sql``
```

Comment:
```sql
explain SELECT
  "id"
  , to_json("created_at") as "createdAt"
  , to_json("updated_at") as "updatedAt"
  , "schema_sdl" as "schemaSDL"
  , "service_name" as "serviceName"
  , "meta"
  , "target_id" as "targetId"
  , "schema_version_id" as "schemaVersionId"
  , "is_success" as "isSuccess"
  , "schema_composition_errors" as "schemaCompositionErrors"
  , "breaking_schema_changes" as "breakingSchemaChanges"
  , "safe_schema_changes" as "safeSchemaChanges"
  , "schema_policy_warnings" as "schemaPolicyWarnings"
  , "schema_policy_errors" as "schemaPolicyErrors"
  , "composite_schema_sdl" as "compositeSchemaSDL"
  , "supergraph_sdl" as "supergraphSDL"
  , "github_check_run_id" as "githubCheckRunId"
  , "github_repository" as "githubRepository"
  , "github_sha" as "githubSha"
  , coalesce("is_manually_approved", false) as "isManuallyApproved"
  , "manual_approval_user_id" as "manualApprovalUserId"
FROM
  "public"."schema_checks"
WHERE
  "target_id" = '97e7ccae-0e08-4fb7-84e0-c589f528816d'
  AND "is_success" = true
ORDER BY
  "target_id" ASC
  , "created_at" DESC
  , "id" DESC
LIMIT 21
```

```
Limit  (cost=0.14..8.17 rows=1 width=530)
  ->  Index Scan using schema_checks_connection_pagination on schema_checks  (cost=0.14..8.17 rows=1 width=530)
        Index Cond: (target_id = '97e7ccae-0e08-4fb7-84e0-c589f528816d'::uuid)
        Filter: is_success
```

We can add another index, to improve the query:

```sql
CREATE INDEX "schema_checks_connection_pagination_with_no_success" ON "schema_checks" (
  "target_id" ASC
  , "created_at" DESC
  , "id" DESC
)
WHERE
  "is_success" = false
;
```

```
Limit  (cost=0.12..8.15 rows=1 width=530)
  ->  Index Scan using schema_checks_connection_pagination_with_no_success on schema_checks  (cost=0.12..8.15 rows=1 width=530)
        Index Cond: (target_id = '97e7ccae-0e08-4fb7-84e0-c589f528816d'::uuid)
```

At the same time when both filters are enabled, we can also make the query use the index by combining the conditions:

```
CREATE INDEX "schema_checks_connection_pagination_with_no_success_and_changes" ON "schema_checks" (
  "target_id" ASC
  , "created_at" DESC
  , "id" DESC
)
WHERE
  "is_success" = false
  AND  (
    jsonb_typeof("safe_schema_changes") = 'array'
    OR jsonb_typeof("breaking_schema_changes") = 'array'
  )
;
```